perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -4192,34 +4214,33 @@ private void setIfNotEmpty(Map<String, String> map, String value, String key) {
 	 * @param form application form
 	 * @return RegistrarModule if present or null
 	 */
-	private RegistrarModule getRegistrarModule(ApplicationForm form) {
-
+	private Set<RegistrarModule> getRegistrarModules(ApplicationForm form) {
 		if (form == null) {
 			// wrong input
 			log.error("[REGISTRAR] Application form is null when getting it's registrar module.");
 			throw new NullPointerException("Application form is null when getting it's registrar module.");
 		}
 
-		if (form.getModuleClassName() != null && !form.getModuleClassName().trim().isEmpty()) {
+		Set<RegistrarModule> modules = new LinkedHashSet<>();
+		if (!form.getModuleClassNames().isEmpty()) {
 
 			RegistrarModule module = null;
 
-			try {
-				log.debug("[REGISTRAR] Attempting to instantiate class: {}", MODULE_PACKAGE_PATH + form.getModuleClassName());
-				module = (RegistrarModule) Class.forName(MODULE_PACKAGE_PATH + form.getModuleClassName()).newInstance();
-				module.setRegistrar(registrarManager);
-			} catch (Exception ex) {
-				log.error("[REGISTRAR] Exception when instantiating module.", ex);
-				return module;
+			for (String regModule : form.getModuleClassNames()) {
+				try {
+					log.debug("[REGISTRAR] Attempting to instantiate class: {}{}", MODULE_PACKAGE_PATH, regModule);
+					module = (RegistrarModule) Class.forName(MODULE_PACKAGE_PATH + regModule)
+						.getDeclaredConstructor().newInstance();
+					module.setRegistrar(registrarManager);
+				} catch (Exception ex) {
+					log.error("[REGISTRAR] Exception when instantiating module. Skipping module {}{}",
+						MODULE_PACKAGE_PATH, regModule, ex);
+				}
+				log.debug("[REGISTRAR] Class {}{} successfully created.", MODULE_PACKAGE_PATH, regModule);
+				modules.add(module);
```

Comment:
Now, since null is assigned to `module` only before the loop, should the exception be thrown (and caught), this line will still execute and add the last assigned module. Which I suppose should not be a problem since it's a set (logs would be doubled), but if it happened on the first iteration, a null value would be added to the list and cause NullPointerExceptions further on.
We need to add `continue` to the end of the catch block, no?
I'm quite tired while reading this so maybe I'm wrong, but try to take a look

Reply:
Well spotted, should now be corrected. Thanks!